### ISSUE TITLE
[Accessibility]  Select element must have an accessible name

### ISSUE THREAD

**Issue by dhamaris** [ec-europa/europa-component-library#4236]
I am using ECL v4.6.3. I am not sure if this problem is solved in the 5.
I was reported by the accessibility team that select elements are not accessible.

Looking at the documentation, there are several ways to achieve it:
[https://dequeuniversity.com/rules/axe/4.10/select-name?application=webdriverjs](https://dequeuniversity.com/rules/axe/4.10/select-name?application=webdriverjs
).
The simplest one would be to have something like this:
`<label for="state">State:</label> <select id="state"></select>
`
In the image you see that the for from label does not match with the id of the select, so it seems that the select remains with no accessible name:
<img width="1415" height="151" alt="Image" src="https://github.com/user-attachments/assets/6a3919b3-ce23-4d52-9bde-4abcd2d9868e" />

Thank you

**Comment by dhamaris** [comment]
Hi, as a workaround i set the id of the select with the extra suffix -toggle, like this in my select.html.twig:

```
{% set select_id = attribute(attributes, 'id') %}
{% set label_id = select_id ~ '-toggle' %}
```

And then in my select have
`<select aria-labelledby="{{ label_id }}">`

**Comment by planctus** [comment]
Hi @dhamaris this seems a problem in the drupal integration of our component, if you check in ECL the label **for** attribute is correctly using the **id** of the select element: https://ec.europa.eu/component-library/v4.11.1/playground/ec/?path=/story/components-forms--select&globals=viewport:responsive

**Comment by dhamaris** [comment]
Hi @planctus! Long time no see :) Thanks for your quick response. I checked and it seems that the element that is being assigned the id is the input button, but the select element remains with no assignment:

<img width="1079" height="244" alt="Image" src="https://github.com/user-attachments/assets/c0efaff6-e81c-4a61-afff-14b8fdd74564" />

I don't find the place in the js where the this.select gets the id with the -toggle suffix, or at least, a labelledby with the id of the label.

Thanks

**Comment by planctus** [comment]
ah, true, indeed this has changed in v5: https://v5-dev--europa-component-library.netlify.app/playground/ec/?path=/story/components-forms--select-multiple&globals=viewport:responsive
The "for" value matches the select id in the multiple select as well.
Unfortunately it is not foreseen at the moment to release in v4, if you already have a workaround keep using it ;)

**Comment by dhamaris** [comment]
Ok, no problem, good thing is was solved in the v5, thanks!
